### PR TITLE
[runtime env] remove timing check on runtime env test

### PR DESCRIPTION
**Pull request by aslonnie**
it is source of flakiness.

### CODE REVIEW

**Comment by aslonnie** [comment]
failure example: https://buildkite.com/ray-project/premerge/builds/12351#018bef66-3c28-4e7c-aca2-01b9e4fa27c4/5085-5240

File: python/ray/tests/test_runtime_env_2.py [hunk 1]
```diff
@@ -62,23 +62,15 @@ def f(self):
     ):
         ray.get(a.f.remote())
 
-    # The second time this runs it should be faster as the error is cached.
-    start = time.time()
     with pytest.raises(RuntimeEnvSetupError, match="ResolvePackageNotFound"):
         ray.get(f.options(runtime_env=bad_env).remote())
 
-    assert (time.time() - start) < (first_time / 2.0)
```
Comment (aslonnie) [inline python/ray/tests/test_runtime_env_2.py:70]: do not quite understand with this needs to be `first_time / 2`

**Comment by aslonnie** [comment]
ping reviewers.

**Review by rkooo567 — APPROVED** [review]

**Comment by SongGuyang** [comment]
@architkulkarni check about this?

File: python/ray/tests/test_runtime_env_2.py (hunk 1 above)
Comment (architkulkarni) [reply python/ray/tests/test_runtime_env_2.py:70]: I think it was just a guess for something in between the uncached speed and the cached speed. @SongGuyang might have more context about this part.

**Review by architkulkarni — APPROVED** [review]
Thanks for finding and fixing this.  When this is merged we can create a Github issue to track the TODO.

**Comment by rkooo567** [comment]
https://github.com/ray-project/ray/issues/41393

**Pull request merged**